### PR TITLE
bug/ICS-122 did not backwards compatible

### PR DESCRIPTION
**Pull request by JGiter**
- Pipe which transforms DID to appropriate form

### CODE REVIEW

**Review by jrhender — COMMENTED** [review]
This looks like a great start! Good call on the pipe 👍 

Just need to set the `chain` in `DID` to `CHAIN_NAME` from config if it is missing, I think 👍

File: src/modules/did/did.types.ts [hunk 1]
```diff
@@ -7,39 +17,33 @@ export class DID {
...
+    this.chain = <Chain>chain;
+    this.id = id;
+    // If an ethr DID is received that doesn't have chain info, then add it.
+    // This is because we want to be backwards compatible with EWF code that didn't include the chain info in the DID
+    // But we want to add the chain info because the DID resolution is actually occurring from a specific chain
+    if (this.chain) {
+      this.did = `did:${this.method}:${this.chain}:${this.id}`;
     } else {
-      throw new Error('Unsupported DID format');
+      this.did = `did:${this.method}:$${this.id}`;
```
Comment (Harasz) [inline src/modules/did/did.types.ts:45]:
```suggestion
      this.did = `did:${this.method}:${this.id}`;
```
??

**Review by Harasz — CHANGES_REQUESTED** [review]
What about other places? I think we are querying DID in a couple of other places. In the latest migration file (1638185287373-AddChainIDToDIDs.ts) you can find all places in database where we are using DIDs.

**Comment by JGiter** [comment]
> What about other places? I think we are querying DID in a couple of other places. In the latest migration file (1638185287373-AddChainIDToDIDs.ts) you can find all places in database where we are using DIDs.

It requires more efforts to make all endpoints backward compatible. For example piping `did` query parameter of claims endpoints requires piping accompanying `IClaimIssuance`, `IClaimRequest` and `IClaimRejection`. I think it is risky to start such big changes during release @jrhender ?

**Review by jrhender — APPROVED** [review]
This looks good.

**Comment by jrhender** [comment]
> It requires more efforts to make all endpoints backward compatible. For example piping did query parameter of claims endpoints requires piping accompanying IClaimIssuance, IClaimRequest and IClaimRejection. I think it is risky to start such big changes during release @jrhender ?

Yes, I think it's better to get this changed merged and then we make further changes if we need to.

**Comment by jrhender** [comment]
Looking at `1638185287373-AddChainIDToDIDs.ts`, I think these are all the spots:
```
    // asset table
    await queryRunner.query(this.getUpUpdateSQL('asset', 'id'));
    await queryRunner.query(this.getUpUpdateSQL('asset', 'owner'));
    await queryRunner.query(this.getUpUpdateSQL('asset', `"offeredTo"`));
    await queryRunner.query(this.getUpUpdateSQL('asset', `"documentId"`));

    // assets_history table
    await queryRunner.query(
      this.getUpUpdateSQL('assets_history', `"emittedBy"`),
    );
    await queryRunner.query(
      this.getUpUpdateSQL('assets_history', `"relatedTo"`),
    );
    await queryRunner.query(this.getUpUpdateSQL('assets_history', `"assetId"`));

    // did_contact table
    await queryRunner.query(this.getUpUpdateSQL('did_contact', 'did'));
    await queryRunner.query(this.getUpUpdateSQL('did_contact', 'created_by'));

    // did_document_entity table
    await queryRunner.query(this.getUpUpdateSQL('did_document_entity', 'id'));

    // role_claim table
    await queryRunner.query(this.getUpUpdateSQL('role_claim', 'requester'));
    await queryRunner.query(this.getUpUpdateSQL('role_claim', 'subject'));
    await queryRunner.query(this.getUpUpdateSQL('role_claim', `"acceptedBy"`));
```
I think we're still not backwards compatible on the `did_contact` and `role_claim`. But I think folks are accessing those endpoints through Switchboard, which has the latest iam-client-lib, so I think we're okay to leave those 🤞

**Comment by JGiter** [comment]
Ok. Will deal with rest endpoints later

**Comment by jrhender** [comment]
@JGiter let's do it in a separate story if necessary, IMO. Your talents can be put to use elsewhere for now.
If @Harasz is okay with it, let's merge this one

**Review by Harasz — APPROVED** [review]
Fair enough, let's merge it

**Pull request merged**